hack/generator/pkg/codegen/storage/property_converter.go

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package storage
+
+import (
+	"fmt"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/pkg/errors"
+)
+
+// PropertyConverter is used to convert the properties of object types as required for storage variants
+type PropertyConverter struct {
+	visitor             astmodel.TypeVisitor // Visitor used to achieve the required modification
+	types               astmodel.Types       // All the types for this group
+	propertyConversions []propertyConversion // Conversion rules to use for properties when creating storage variants
+}
+
+// NewPropertyConverter creates a new property converter for modifying object properties
+func NewPropertyConverter(types astmodel.Types) *PropertyConverter {
+	result := &PropertyConverter{
+		types: types,
+	}
+
+	result.propertyConversions = []propertyConversion{
+		result.preserveKubernetesResourceStorageProperties,
+		result.defaultPropertyConversion,
+	}
+
+	result.visitor = astmodel.TypeVisitorBuilder{
+		VisitEnumType:      result.useBaseTypeForEnumerations,
+		VisitValidatedType: result.stripAllValidations,
+		VisitTypeName:      result.shortCircuitNamesOfSimpleTypes,
+	}.Build()
+
+	return result
+}
+
+// ConvertProperty applies our conversion rules to a specific property
+func (p *PropertyConverter) ConvertProperty(property *astmodel.PropertyDefinition) (*astmodel.PropertyDefinition, error) {
+
+	for _, conv := range p.propertyConversions {
+		prop, err := conv(property)
+		if err != nil {
+			// Something went wrong, return the error
+			return nil, err
+		}
+		if prop != nil {
+			// We have the conversion we need, return it promptly
+			return prop, nil
+		}
+	}
+
+	return nil, fmt.Errorf("failed to find a conversion for property %v", property.PropertyName())
+}
+
+// stripAllValidations removes all validations
+func (p *PropertyConverter) stripAllValidations(
+	this *astmodel.TypeVisitor, v *astmodel.ValidatedType, ctx interface{}) (astmodel.Type, error) {
+	// strip all type validations from storage property types
+	// act as if they do not exist
+	return this.Visit(v.ElementType(), ctx)
+}
+
+// useBaseTypeForEnumerations replaces an enumeration with its underlying base type
+func (p *PropertyConverter) useBaseTypeForEnumerations(
+	tv *astmodel.TypeVisitor, et *astmodel.EnumType, ctx interface{}) (astmodel.Type, error) {
+	return tv.Visit(et.BaseType(), ctx)
+}
+
+// shortCircuitNamesOfSimpleTypes redirects TypeNames that reference resources or objects into our
+// storage namespace, and replaces TypeNames that point to simple types (enumerations or
+// primitives) with their underlying type.
+func (p *PropertyConverter) shortCircuitNamesOfSimpleTypes(
```

Comment:
Moving comment from last review over here:
Is the main reason that you need to do this for things like validated types where they had an alias before but since you're dropped validations there's no need for an alias now?

Reply:
expanding: It might be helpful to explain (in this methods comment maybe?) the purpose of this function in addition to what it does.

Reply:
Improve the comments